source/common/access_log/access_log_formatter.cc

```diff
@@ -110,36 +112,46 @@ std::vector<FormatterPtr> AccessLogFormatParser::parse(const std::string& format
         current_token = "";
       }
 
-      size_t command_end_position = format.find('%', pos + 1);
-      if (command_end_position == std::string::npos) {
-        throw EnvoyException(fmt::format(
-            "Incorrect configuration: {}. Expected end of operation '%', around position {}",
-            format, pos));
+      std::smatch m;
+      std::string search_space = format.substr(pos);
+      if (!(std::regex_search(search_space, m, command_w_args_regex) || m.position() == 0)) {
+        throw EnvoyException(
+            fmt::format("Incorrect configuration: {}. Couldn't find valid command at position {}",
+                        format, pos));
       }
-      std::string token = format.substr(pos + 1, command_end_position - (pos + 1));
+
+      const std::string match = m.str(0);
+      const std::string token = match.substr(1, match.length() - 2);
+      pos += 1;
+      int command_end_position = pos + token.length();
 
       if (token.find("REQ(") == 0) {
         std::string main_header, alternative_header;
         absl::optional<size_t> max_length;
-        const size_t start = 4;
 
-        parseCommand(token, start, main_header, alternative_header, max_length);
+        parseCommand(token, REQ_PARAM_START, main_header, alternative_header, max_length);
```

Comment:
I think we need to reconcile https://github.com/envoyproxy/envoy/pull/2895 with this as well. @glicht FYI. Can you folks decide which order you'd prefer we merge in?

Reply:
this formatting PR is a blocker for us so we'd like this wrapped up soon but looking at #2895 I don't think it would take long to modify my PR for @glicht's changes - I'm fine with whichever lands first.

Reply:
I am also fine with whichever lands first. The merge shouldn't be too complicated.